src/handlers/bot_pull_requests.rs

```diff
@@ -0,0 +1,66 @@
+use indexmap::IndexSet;
+use std::sync::atomic::AtomicBool;
+use std::sync::{LazyLock, Mutex};
+
+use crate::github::{IssueRepository, IssuesAction, PrStatus};
+use crate::{github::Event, handlers::Context};
+
+pub(crate) async fn handle(ctx: &Context, event: &Event) -> anyhow::Result<()> {
+    let Event::Issue(event) = event else {
+        return Ok(());
+    };
+    if event.action != IssuesAction::Opened {
+        return Ok(());
+    }
+    if !event.issue.is_pr() {
+        return Ok(());
+    }
+
+    // avoid acting on our own open events, otherwise we'll infinitely loop
+    if event.sender.login == ctx.username {
+        return Ok(());
+    }
+
+    // If it's not the github-actions bot, we don't expect this handler to be needed. Skip the
+    // event.
+    if event.sender.login != "github-actions" {
```

Comment:
I'm not sure if this will work, an user with such login doesn't seem to exist? https://github.com/github-actions

I tried to run this on https://github.com/rust-lang/miri/pull/4070:
![image](https://github.com/user-attachments/assets/290acea9-f87a-4ac6-bb02-138723a537be)

It looks like the login is `app/github-actions`. Ideally we should also check if it's a bot.

Reply:
Hm, I'm not sure if we get is_bot = true in the `sender` field, I'd have to go look at events we've seen. I've updated to app/github-actions, and we can revisit if this doesn't work in practice IMO.